java/src/jmri/jmrix/mqtt/MqttAdapter.java

```diff
@@ -257,27 +257,43 @@ public MqttClient getMQttClient() {
         return (mqttClient);
     }
 
+    private void tryToReconnect() {
+        log.warn("Try to reconnect");
+        try {
+            if ( getOptionState(MQTT_USERNAME_OPTION) != null
+                    && ! getOptionState(MQTT_USERNAME_OPTION).isEmpty()) {
+                mqttClient.connect(getMqttConnectionOptions());
+            } else {
+                mqttClient.connect();
+            }
+            log.warn("Succeeded to reconnect");
+
+            mqttClient.setCallback(this);
+            for (String t : mqttEventListeners.keySet()) {
+                mqttClient.subscribe(t);
+            }
+        } catch (MqttException ex) {
+            log.error("Unable to reconnect", ex);
+            scheduleReconnectTimer();
+        }
+    }
+
+    private void scheduleReconnectTimer() {
+        jmri.util.TimerUtil.scheduleOnLayoutThread(new java.util.TimerTask() {
+            @Override
+            public void run() {
+                tryToReconnect();
+            }
+        }, 100);
```

Comment:
Fixed.

Reply:
Will this put a new Message in the system console every half second, or just a single message when it loses connection and starts to retry? Single message of “Attempting to reconnect every 1/2 second” would be sufficient since  a connection message would show up if/when it reconnects.  In my own case, may take me several minutes to see what is happening and restart the broker so don’t want hundreds of messages in the console.  Thanks.

Reply:
I have updated the code so that you only get messages in the log when the connection disconnects and when it reconnects.

However, if you throw a turnout, you will still get errors in the log since JMRI cannot forward that message to the broker.

Reply:
Sounds fine.  Thanks.